test/integration/test_dataplane_scale_subresource.go

```diff
@@ -105,4 +85,82 @@ func TestDataPlaneScaleSubresource(t *testing.T) {
 
 	t.Log("verifying dataplane scales to 3 replicas")
 	require.Eventually(t, testutils.DataPlaneHasNReadyPods(t, GetCtx(), dataplaneName, clients, 3), testutils.GatewayReadyTimeLimit, testutils.ObjectUpdateTick)
+
+	// Get the current ReplicaSets before restart
+	replicaSetList := &appsv1.ReplicaSetList{}
+	err = clients.MgrClient.List(GetCtx(), replicaSetList,
+		client.InNamespace(namespace.Name),
+		client.MatchingLabels(deployment.Spec.Selector.MatchLabels))
+	require.NoError(t, err)
+	require.NotEmpty(t, replicaSetList.Items, "Expected at least one ReplicaSet")
+
+	// Record the names of existing ReplicaSets
+	existingRSNames := make(map[string]struct{})
+	for _, rs := range replicaSetList.Items {
+		existingRSNames[rs.Name] = struct{}{}
+	}
+
+	t.Log("simulating kubectl rollout restart by adding restart annotation")
+	deploymentCopy := deployment.DeepCopy()
+	if deploymentCopy.Spec.Template.Annotations == nil {
+		deploymentCopy.Spec.Template.Annotations = make(map[string]string)
+	}
+	deploymentCopy.Spec.Template.Annotations["kubectl.kubernetes.io/restartedAt"] = time.Now().Format(time.RFC3339)
+
+	err = clients.MgrClient.Patch(GetCtx(), deploymentCopy, client.MergeFrom(&deployment))
+	require.NoError(t, err)
+
+	t.Log("waiting for new ReplicaSet to be created")
+
+	var newReplicaSet *appsv1.ReplicaSet
+	require.Eventually(t, func() bool {
+		newReplicaSetList := &appsv1.ReplicaSetList{}
+		if err := clients.MgrClient.List(GetCtx(), newReplicaSetList,
+			client.InNamespace(namespace.Name),
+			client.MatchingLabels(deployment.Spec.Selector.MatchLabels)); err != nil {
+			t.Logf("Error listing ReplicaSets: %v", err)
+			return false
+		}
+
+		// Find the new ReplicaSet (one that wasn't in the original list)
+		for i := range newReplicaSetList.Items {
+			rs := &newReplicaSetList.Items[i]
+			if _, exists := existingRSNames[rs.Name]; !exists {
+				newReplicaSet = rs
+				return true
+			}
+		}
+		return false
+	}, testutils.GatewayReadyTimeLimit, testutils.ObjectUpdateTick, "Failed to find new ReplicaSet after rollout restart")
```

Comment:
WDYT about adding a helper func which will for example return a `ReplicaSet` which is newer than the provided `time.Time`? This helper could fail the test (using the provided `*testing.T`) if there's more than 1.

I believe this could improve the readability of this test. This would also prevent the need for `existingRSNames`.

Reply:
Addressed and removed the `existingRSNames`